pkg/nsm/interfacename/generator.go

```diff
@@ -51,3 +52,30 @@ func (rg *RandomGenerator) Generate(prefix string, maxLength int) string {
 	rg.usedNames[randomName] = struct{}{}
 	return randomName
 }
+
+type CounterGenerator struct {
+	mu        sync.Mutex
+	usedNames map[int]struct{}
+}
+
+// TODO: make sure the generated name is within range (probably the prefix length should be limited as well)
+func (cg *CounterGenerator) Generate(prefix string, maxLength int) string {
+	cg.mu.Lock()
+	defer cg.mu.Unlock()
+	if cg.usedNames == nil {
+		cg.usedNames = make(map[int]struct{})
+	}
+	selected := 0
+	length := maxLength - len(prefix)
+	length *= 10
+	length -= 1
```

Comment:
Imho, max allowed decimal counter value should be (10^length)-1

Also, just noticed that MAX_INTERFACE_NAME_LENGTH is set to 16, while that is only true if the terminating null is included. So either change MAX_INTERFACE_NAME_LENGTH to 15, or use (10^(length-1))-1 as max counter value.

Reply:
Fixed, thank you